pkg/ocmagenthandler/ocmagenthandler_secret.go

```diff
@@ -29,6 +30,30 @@ func buildOCMAgentAccessTokenSecret(accessToken []byte, ocmAgent ocmagentv1alpha
 	return secret
 }
 
+func fleetOCMAgentSecret(ocmAgent ocmagentv1alpha1.OcmAgent) corev1.Secret {
```

Comment:
A few questions come to mind regarding this.

The first is: is it necessary at all? We'll already have a `Secret` distributed to the cluster via app-interface, so all OAO should really need to tell a fleet-mode OA is the name/namespace of the Secret.

The second is: if we deem that it is necessary, we still shouldn't need to do any of the 'environment-deciding stuff'. One of the keys in the secret that is supplied via app-interface will be the URL that the token is used for (`OA_OCM_URL`).

Reply:
Yes, I have the same concerns. The purpose I added this part is because we have `ensureAccessTokenSecret` for OSD/ROSA, I thought we should do something at least to check the known value and whether it is expected. We still need all the values to supply from the app-interface because we do not build the secret here.